### PR TITLE
[PEPPER-309] [Playwright] Misc minor updates

### PR DESCRIPTION
**Pull request by aweng98**
Few general improvement for better test performance and stability:

- Update playwright.config.ts: Disable fullyParallel.
- Update `waitUtil` value to `load` from `domcontentloaded` based off latest Playwright release note
- Intercept and abort Google analytics requests

### CODE REVIEW

File: playwright-e2e/authentication/auth-singular.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export async function fillEmailPassword(
   const passwordInput = page.locator('input[type="password"]');
   await passwordInput.fill(password);
 
-  const navPromise = waitForNavigation ? page.waitForNavigation({ waitUntil: 'domcontentloaded' }) : Promise.resolve();
+  const navPromise = waitForNavigation ? page.waitForNavigation({ waitUntil: 'load' }) : Promise.resolve();
```
Comment (aweng98) [inline playwright-e2e/authentication/auth-singular.ts:18]: Since I will update Playwright version soon, I'm going to make [this](https://playwright.dev/docs/release-notes#behavior-change) mentioned change now for all `waitUntil`.

File: playwright-e2e/fixtures/singular-fixture.ts [hunk 2]
```diff
@@ -7,6 +7,8 @@ type Fixtures = {
   homePage: SingularHomePage;
 };
 
+const REQUEST_EXCLUDES = ['google-analytics'];
```
Comment (aweng98) [inline playwright-e2e/fixtures/singular-fixture.ts:10]: Test generated analytics are not useful at all.

File: playwright-e2e/pages/singular/enrollment/pre-screening-page.ts [hunk 3]
```diff
@@ -59,10 +59,11 @@ export default class PreScreeningPage {
 
   async checkReCaptcha(): Promise<void> {
     const iframe = this.page.frameLocator('css=iframe[title="reCAPTCHA"]');
+    await iframe.locator('css=span[role="checkbox"]').waitFor({ state: 'visible' });
     await iframe.locator('css=span[role="checkbox"]').dispatchEvent('click');
     await iframe
       .locator('.recaptcha-checkbox-spinner[style*="animation-play-state: running;"]')
-      .waitFor({ state: 'hidden' });
+      .waitFor({ state: 'hidden', timeout: 30 * 1000 });
```
Comment (aweng98) [inline playwright-e2e/pages/singular/enrollment/pre-screening-page.ts:66]: Default timeout is 10 sec. Increase to 30 sec because it had failed once with timeout error on localhost.

File: playwright-e2e/playwright.config.ts [hunk 4]
```diff
@@ -28,8 +28,7 @@ const testConfig: PlaywrightTestConfig = {
     toMatchSnapshot: {
       // Account for minor difference in text rendering and resolution between headless and headed mode
       threshold: 0.3,
-      maxDiffPixelRatio: 0.2,
-      maxDiffPixels: 200
```
Comment (aweng98) [inline playwright-e2e/playwright.config.ts:32]: Specify all three rules meant that all rules must be met in order for the screenshot comparison to pass. `maxDiffPixels: 200` is not needed because it could contradict with other two rules.

File: playwright-e2e/playwright.config.ts [hunk 5]
```diff
@@ -38,7 +37,7 @@ const testConfig: PlaywrightTestConfig = {
     }
   },
   /* Run tests in files in parallel */
-  fullyParallel: true,
+  // fullyParallel: true,
```
Comment (aweng98) [inline playwright-e2e/playwright.config.ts]: I don't think we want to keep this, run different tests in same suite in parallel. It could potentially cause tests to be flaky.

Comment (bskinner) [reply playwright-e2e/playwright.config.ts]: I agree with disabling parallel testing. Instead of leaving a commented out line there, can it be either removed entirely, or explicitly set to `false`?

File: playwright-e2e/authentication/auth-singular.ts (hunk 1 above)
Comment (bskinner) [reply playwright-e2e/authentication/auth-singular.ts:18]: Will this introduce any breakage in the tests until the Playwright version is updated?

Comment (aweng98) [reply playwright-e2e/authentication/auth-singular.ts:18]: I don't think it will. I didn't see any failures when running tests locally. Playwright dev team is pretty good with tool quality and not introduce breaking changes or new bugs.

File: playwright-e2e/playwright.config.ts (hunk 5 above)
Comment (aweng98) [reply playwright-e2e/playwright.config.ts]: Good call. I'll set it to `false`.

File: playwright-e2e/fixtures/singular-fixture.ts [hunk 6]
```diff
@@ -15,6 +17,14 @@ type Fixtures = {
...
 // This fixture runs per test when called. Removes duplicated code in every test.
 const fixture = baseTest.extend<Fixtures>({
+  page: async ({ baseURL, page }, use) => {
+    await page.route('**/*', (route) => {
+      return REQUEST_EXCLUDES.some((urlPart) => route.request().url().includes(urlPart))
+        ? route.abort()
+        : route.continue();
+    });
+    await use(page);
+  },
```
Comment (bskinner) [inline playwright-e2e/fixtures/singular-fixture.ts:27]: Could you summarize what's going on here? I see it's performing filtering based on some URL pattern, but I'm unsure what the intent of the filtering is.

File: playwright-e2e/pages/singular/enrollment/pre-screening-page.ts (hunk 3 above)
Comment (bskinner) [reply playwright-e2e/pages/singular/enrollment/pre-screening-page.ts:66]: Could the reasoning for using `30s` be added a comment so future devs aren't left wondering why this is the particular magic value to be used?

**Review by bskinner — APPROVED** [review]
Looks good to me. Only thing I would ask is that you add a high-level summary to the description so it's easier to see what this PR changes at a glance- `General improvement` is a vague description.

File: playwright-e2e/fixtures/singular-fixture.ts (hunk 6 above)
Comment (aweng98) [reply playwright-e2e/fixtures/singular-fixture.ts:27]: General good test automation practice, abort seding non-product requests. A little bit, it improves page performance, decrease network bandwidth usage.

For now, intercept and abort all Google `https://www.google-analytics.com/` requests because analytic data generated by automation tests are generally not meaningful and useful.

Comment (bskinner) [reply playwright-e2e/fixtures/singular-fixture.ts:27]: Thank you!

File: playwright-e2e/pages/singular/enrollment/pre-screening-page.ts (hunk 3 above)
Comment (Rinchan20) [reply playwright-e2e/pages/singular/enrollment/pre-screening-page.ts:66]: +1 to Blake's comment

**Review by Rinchan20 — APPROVED** [review]

File: playwright-e2e/pages/singular/enrollment/pre-screening-page.ts (hunk 3 above)
Comment (aweng98) [reply playwright-e2e/pages/singular/enrollment/pre-screening-page.ts:66]: No particular reason. 30 sec is a common value for timeout in test automation, normally based on most network requests duration times. It's not long and not short.

**Comment by aweng98** [comment]
Thank you all.

File: playwright-e2e/pages/singular/enrollment/pre-screening-page.ts (hunk 3 above)
Comment (bskinner) [reply playwright-e2e/pages/singular/enrollment/pre-screening-page.ts:66]: Sorry- to clarify, I meant as a comment in the code itself (especially if the default is 10s)

Comment (aweng98) [reply playwright-e2e/pages/singular/enrollment/pre-screening-page.ts:66]: My fault. I should have explained as a comment. I'll add it in a follow-up PR. Also, I'll add comments for few other places where 30 sec are also used.

Comment (bskinner) [reply playwright-e2e/pages/singular/enrollment/pre-screening-page.ts:66]: No worries- thank you for the details!
